### ISSUE TITLE
Multiple authorization support not being recognized from CLI

### ISSUE THREAD

**Issue by RussMax783** [aws-amplify/amplify-cli#3570]
**Describe the bug**
When trying to set up a graphql api that allows a model to be publicly accessible with an API key but also have @auth restrictions with cognito user pools, it only respects the first/default option selected. In the `backend-config.json` this is the result of running the steps below: 
```
...
              "authConfig": {
					"additionalAuthenticationProviders": [],
					"defaultAuthentication": {
						"authenticationType": "API_KEY",
						"apiKeyConfig": {
							"description": "public api",
							"apiKeyExpirationDays": 7
						}
					}
				}
...
```

Manually updating App Sync from the console to have the additional authorization providers only partially works, because you still wont be able to update your schema to have the public and owner rules in the @auth directive. The CLI will throw `@auth directive with 'userPools' provider found, but the project has no Cognito User Pools authentication provider configured.` 

**Amplify CLI Version**
4.13.4

**To Reproduce**
from the cli:
? Please select from one of the below mentioned services: GraphQL
? Choose the default authorization type for the API API key
? Enter a description for the API key: public api
? After how many days from now the API key should expire (1-365): 7
? Do you want to configure advanced settings for the GraphQL API Yes, I want to make some additional changes.
? Configure additional auth types? Yes
? Choose the additional authorization types you want to configure for the API 
	(chose cognito user pool) 
? Configure conflict detection? No

**Expected behavior**
I would expect the CLI to be able to add the secondary auth provider to the API and allow for multiple auth rules.

**Desktop (please complete the following information):**
 - OS: Mac
 - Node Version: v12.6

From blog posts and docs, i have found a couple resources explaining this functionality:
- https://dev.to/dabit3/12-new-features-of-aws-amplify-4i8h
- https://aws-amplify.github.io/docs/cli-toolchain/graphql#allowed-authorization-mode-vs-provider-combinations
- https://dev.to/robertbroersma/querying-owned-posts-vs-all-posts-in-aws-amplify-35jl

**Comment by nikhname** [comment]
Hi @RussMax783 in order to add the secondary auth provider you would need to go through the update api flow, running ```amplify update api```. Please feel free to comment if you have further questions

**Comment by RussMax783** [comment]
@nikhname thanks for the response. Even when going through the update it does not add the secondary. I've tried re-installing the CLI, installing previous versions of the CLI, and started up a project from scratch. In the `backend-config.json`, it doesn't add anything to the `additionalAuthenticationProviders` field. Even if I manually put some values in there, no change is detected when i run `amplify status`.

**Comment by kjetilge** [comment]
Same thing here. After updating api with adding api key, ```amplify status``` reports ```No Change```

**Comment by kaustavghosh06** [comment]
@kjetilge @RussMax783 If you install the latest version of the CLI - you can use `amplify push --force` to push changes for just your auth settings to the cloud.

**Comment by RussMax783** [comment]
I updated my backend-config file and tried doing a force push. In the terminal it gives me a GraphQL API KEY, but things are still in a bad state. I can't add @auth directive to my models to allow for public access as it still give me an error when i try and update. After trying to run an amplify command the backend-config file went back to only having the one `defaultAuthentication` and removed the api key setting I added in `additionalAuthenticationProviders`
